packages/koa-error-handler/index.js

```diff
@@ -1,13 +1,19 @@
-module.exports = () => async (ctx, next) => {
+module.exports = ({ renderHtml } = {}) => async (ctx, next) => {
   try {
     await next();
   } catch (err) {
+    const contentType = ctx.accepts('html', 'json');
     ctx.status = 500;
-    ctx.body = {
-      statusCode: 500,
-      error: 'Internal Server Error',
-      message: 'An internal server error occurred'
-    };
+    if (typeof renderHtml === 'function' && contentType === 'html') {
+      ctx.set('Content-Type', 'text/html; charset=utf-8');
+      await renderHtml(ctx);
```

Comment:
I think we should set the ctx.body with the return of the renderHtml function.

```
ctx.body = await renderHtml();
```

like we do for JSON response.

Also we should add try catch around it and if it fail send ugly HTML ?

Reply:
The advantage of passing 'ctx' is that you can easily use @sigfox/koa-nunjucks
```
app.use(errorHandler({ renderHtml: ctx => ctx.render('500') }));
```
But it's true that it would be better as you say, I'm going to see if we can find a solution to avoid duplicating the code of 'koa-nunjucks'.

Reply:
I found a good solution by modifying 'koa-nunjucks'.
Adding 'nunjucksRender', that returns a simple rendering as a string.
You can see it on the latest commits